wallet/src/actors/worker/methods.rs

```diff
@@ -0,0 +1,263 @@
+use std::convert::TryFrom;
+
+use jsonrpc_core as rpc;
+use rayon::prelude::*;
+use serde_json::json;
+
+use super::*;
+use crate::{account, crypto, db::Database as _, model, params};
+
+impl Worker {
+    pub fn start(concurrency: usize, db: Arc<rocksdb::DB>, params: params::Params) -> Addr<Self> {
+        let engine = types::SignEngine::signing_only();
+        let wallets = Arc::new(repository::Wallets::new(db::PlainDb::new(db.clone())));
+
+        SyncArbiter::start(concurrency, move || Self {
+            db: db.clone(),
+            wallets: wallets.clone(),
+            params: params.clone(),
+            rng: rand_os::OsRng,
+            engine: engine.clone(),
+        })
+    }
+
+    pub fn run_rad_request(&self, request: types::RADRequest) -> Result<types::RadonTypes> {
```

Comment:
I think all the data request logic should not be implemented here, the witnet_radon crate should expose a `run_rad_request` method which would return a `Future<Item=RadonTypes, Error=RadonError>`.

Reply:
I agree, although the implementation @aesedepece and @Tommytrg had done is not async so it doesn't work with futures, maybe we could open an issue to address later, what do you think @mariocao ?